src/server/api/circuits.go

```diff
@@ -95,6 +95,43 @@ func circuitLoadHandler(m interfaces.CircuitStorageInterfaceFactory, c *gin.Cont
 	c.JSON(http.StatusOK, circuit)
 }
 
+func circuitUpdateHandler(m interfaces.CircuitStorageInterfaceFactory, c *gin.Context, userId model.UserId) {
+	storageInterface := m.CreateCircuitStorageInterface()
+
+	circuitId := c.Param("id")
+
+	needToCreate := false
+	circuit := storageInterface.LoadCircuit(circuitId)
+	if circuit == nil {
+		needToCreate = true
+	}
+	newCircuit, err := parseCircuitRequestData(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	if needToCreate {
```

Comment:
```suggestion
	if circuit := storageInterface.LoadCircuit(circuitId); circuit != nil {
```
If you flip the conditional you can use this neat syntax feature.  We probably should be parsing the request body before loading the circuit anyway too.

Reply:
Yes, that's why I didn't write Fixes #504 at the end. I will refactor the code and maybe commit tomorrow.